scripts/segmentation_HALO-20240909a.md

```diff
@@ -0,0 +1,332 @@
+plt.plot(meteor_track.lon, meteor_track.lat, c="C4", ls=":", label="METEOR track")
+plt.xlabel("longitude / °")
+plt.ylabel("latitude / °")
+plt.legend();
+```
+
+## Interactive plots
+
+```python
+ds["alt"].hvplot()
+```
+
+```python
+ds["roll"].hvplot()
+```
+
+```python
+ds["heading"].hvplot()
+```
+
+## Segments
+
+defined as a tuple of time slice (`start`, `end`) , segment `kind`, `name`, `remarks`.
+
+* in case of irregularities within a circle, 1 sec before the first and after the last sonde are chosen as start and end times
+* use the list of `remarks` to state any deviations, also with respective times
+
+Alternatively, you can also define the segments as dictionaries which also allows to add further attributes to single segments, e.g. a `radius` to a `circle` segment. At the end of the following code block all segments will be normalized by the `parse_segments` function.
+
+```python
+sl1 = (
+    slice("2024-09-09T11:54:30", "2024-09-09T12:12:21"),
+    ["straight_leg", "ascent"],
+    "climbing and heading east"
+)
+
+sl2 = (
+    slice("2024-09-09T12:12:21", "2024-09-09T13:47:15"),
+    ["straight_leg"],
+    "heading east at constant alt",
+    ["irregularity: roll angle deviations around 12:22:44, 12:40:47, 12:54:38, and 13:00:46"]
+)
+
+c1 = (
+    slice("2024-09-09T13:49:10", "2024-09-09T14:45:24"),
+    ["circle"],
+    "circle south",
+)
+
+slc1_1 = (
+    slice("2024-09-09T14:50:04", "2024-09-09T14:53:18"),
+    ["straight_leg", "ascent", "ec_track"],
+    "climbing while crossing circle south on EC track"
+)
+
+slc1_2 = (
+    slice("2024-09-09T15:01:57", "2024-09-09T15:11:40"),
+    ["straight_leg", "ec_track"],
+    "leg crossing circle south on EC track"
+)
+
+c2 = (
+    slice("2024-09-09T15:13:57", "2024-09-09T16:09:04"),
+    ["circle"],
+    "circle mid",
+)
+
+# should we even give it the "circle" attribute?
+c_climb = (
+    slice("2024-09-09T16:09:12", "2024-09-09T16:24:16"),
+    ["circle", "ascent"],
+    "climbing circle and holding pattern to wait for EC",
+)
+
+ec = (
+    slice("2024-09-09T16:26:31", "2024-09-09T17:19:47"),
+    ["straight_leg", "ec_track", "ec_underpass"],
+    "meeting EC",
+)
+
+ec_south = (
+    slice("2024-09-09T17:26:25", "2024-09-09T17:32:34"),
+    ["straight_leg", "ec_track"],
+    "leg southward on EC track"
+)
+
+c3 = (
+    slice("2024-09-09T17:35:00", "2024-09-09T18:30:50"),
+    ["circle"],
+    "circle north",
+    ["irregularity: early start of circle due to dropsondes, constant roll angle after 17:35:19"]
+)
+
+sl3 = (
+    slice("2024-09-09T18:33:32", "2024-09-09T18:55:04"),
+    ["straight_leg"],
+    "leg heading west"
+)
+sl4 = (
+    slice("2024-09-09T18:55:04", "2024-09-09T18:59:15"),
+    ["straight_leg", "descent"],
+    "leg heading west and descending"
+)
+sl5 = (
+    slice("2024-09-09T18:59:15", "2024-09-09T20:14:03"),
+    ["straight_leg"],
+    "leg heading west before landing"
+)
+
```

Comment:
Added a couple segments because I felt they were long and clean enough to be included.
```suggestion
sl6 = (
    slice("2024-09-09T20:15:49", "2024-09-09T20:20:59"),
    ["straight_leg", "descent"],
    "leg descending towards Barbados"
)

sl7 = (
    slice("2024-09-09T20:22:03", "2024-09-09T20:24:42"),
    ["straight_leg", "descent"],
    "leg descending to Barbados"
)

sl8 = (
    slice("2024-09-09T20:33:08", "2024-09-09T20:39:59"),
    ["straight_leg", "descent"],
    "leg descending to airport"
)

sl9 = (
    slice("2024-09-09T20:42:47", "2024-09-09T20:46:32"),
    ["straight_leg", "descent"],
    "final descent to runway",
    ["irregularity: turbulence near surface with roll angle deviations up to +-2.3 degree"]
)
```